### PR TITLE
[onert] Using mmap and munmap during model load

### PR DESCRIPTION
**Pull request by periannath**
This commit uses mmap and munmap during model load to reduce model load time
  - Loads model file using mmap at base_loader
  - Unmaps whole model file after model load
  - Additional memory allocated for model file is removed
    - vector<char> `_buffer` is replaced to char* `_base`

Signed-off-by: JiHwan Yeo <jihwan.yeo@samsung.com>

---

For issue : #2077
From draft #2097

### CODE REVIEW

File: runtime/onert/frontend/base_loader/include/base_loader.h [hunk 1]
```diff
@@ -175,30 +182,32 @@ template <typename LoaderDomain, typename SpecificLoader> class BaseLoader
...
-  _verifier = std::make_unique<Verifier>(reinterpret_cast<const std::uint8_t *>(_buffer.data()),
-                                         _buffer.size());
+  _verifier = std::make_unique<Verifier>(reinterpret_cast<const std::uint8_t *>(_base), size);
 
   loadModel();
+
+  // Unmap mapped region for model file
+  if (munmap(_base, size) == -1)
+  {
+    VERBOSE(BASE_LOADER) << "munmap failed" << std::endl;
```
Comment (ragmani) [inline runtime/onert/frontend/base_loader/include/base_loader.h:209]: If `munmap` fails, is there no way to unload? It is curious.

File: runtime/onert/frontend/base_loader/include/base_loader.h [hunk 2]
```diff
@@ -175,30 +182,32 @@ template <typename LoaderDomain, typename SpecificLoader> class BaseLoader
...
+  if (fstat(fd, &file_stat) != 0)
+  {
+    throw std::runtime_error("Fstat failed or file " + std::string(file_path) +
+                             " is not a regular file");
+  }
+  int size = file_stat.st_size;
 
-  stream.close();
+  // Map model file into memory region
+  _base = static_cast<char *>(mmap(NULL, size, PROT_READ, MAP_PRIVATE, fd, 0));
```
Comment (ragmani) [inline runtime/onert/frontend/base_loader/include/base_loader.h:200]: Why didn't you use _base as local variable?

Comment (periannath) [reply runtime/onert/frontend/base_loader/include/base_loader.h:200]: @ragmani Following PR will use that value in `loadOperand` funtion to partially unmaps memory region for weight values [ed2e01f](https://github.com/Samsung/ONE/pull/2097/commits/ed2e01f35382bed2e5a8e3abeab10de5eff0bf37)

File: runtime/onert/frontend/base_loader/include/base_loader.h (hunk 1 above)
Comment (periannath) [reply runtime/onert/frontend/base_loader/include/base_loader.h:209]: @ragmani AFAIK, `munmap` is the only way to free mmaped region. If it fails, mmaped region will be remained at the memory.

File: runtime/onert/frontend/base_loader/include/base_loader.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (YongseopKim) [reply runtime/onert/frontend/base_loader/include/base_loader.h:200]:
```c++
if (_base == MAP_FAILED) {
  throw std::runtime_error("mmap failed by " + errno);
}
```

[the remark this answers is not in the document]
Comment (periannath) [reply runtime/onert/frontend/base_loader/include/base_loader.h:200]: @YongseopKim Thank you! I'll update it soon.

Comment (ragmani) [reply runtime/onert/frontend/base_loader/include/base_loader.h:200]: I see. Thanks.

File: runtime/onert/frontend/base_loader/include/base_loader.h (hunk 1 above)
Comment (ragmani) [reply runtime/onert/frontend/base_loader/include/base_loader.h:214]: Okay, Thanks for your information.

**Pull request merged**